likecoin/js/admin/likecoin_metabox.js

```diff
@@ -36,55 +64,212 @@ async function onISCNCallback(event) {
-  e.preventDefault();
+  if (e) e.preventDefault();
   const {
-    title,
-    ipfsHash,
-    tags,
-    url,
+    title, mattersIPFSHash, arweaveIPFSHash, tags, url, arweaveId,
   } = lcPostInfo;
   const { siteurl } = wpApiSettings;
-  if (!ipfsHash) return;
-  const titleString = encodeURIComponent(title);
-  const tagsArray = tags || [];
-  const tagsString = tagsArray.join(',');
-  const urlString = encodeURIComponent(url);
-  const redirectString = encodeURIComponent(siteurl);
-  const likeCoISCNWidget = `https://like.co/in/widget/iscn?fingerprint=${ipfsHash}&publisher=matters&title=${titleString}&tags=${tagsString}&opener=1&url=${urlString}&redirect_uri=${redirectString}`;
-  window.open(likeCoISCNWidget, '_blank', 'menubar=no,location=no,width=576,height=768');
-  window.addEventListener('message', onISCNCallback, false);
+  lcPostInfo.arweaveAndISCNUploadStatus = 'uploading to ISCN...';
+  try {
+    if (!mattersIPFSHash && !arweaveIPFSHash && !arweaveId) {
+      throw new Error('NO_IPFS_HASH_NOR_ARWEAVE_ID_FOUND');
+    }
+    const titleString = encodeURIComponent(title);
+    const tagsArray = tags || [];
+    const tagsString = tagsArray.join(',');
+    const urlString = encodeURIComponent(url);
+    const redirectString = encodeURIComponent(siteurl);
+    const fingerprints = [];
+    if (mattersIPFSHash) {
+      const mattersIPFSHashFingerprint = `ipfs://${mattersIPFSHash}`;
+      fingerprints.push(mattersIPFSHashFingerprint);
+    }
+    if (arweaveIPFSHash) {
+      const arweaveIPFSHashFingerprint = `ipfs://${arweaveIPFSHash}`;
+      fingerprints.push(arweaveIPFSHashFingerprint);
+    }
+    if (arweaveId) {
+      const arweaveFingerprint = `ar://${arweaveId}`;
+      fingerprints.push(arweaveFingerprint);
+    }
+    const fingerprint = fingerprints.join(',');
+    const likeCoISCNWidget = `https://like.co/in/widget/iscn?fingerprint=${fingerprint}&publisher=matters&title=${titleString}&tags=${tagsString}&opener=1&blocking=1&url=${urlString}&redirect_uri=${redirectString}`;
+    window.open(
+      likeCoISCNWidget,
+      'likeCoISCNWindow',
+      'menubar=no,location=no,width=576,height=768',
+    );
+    window.addEventListener('message', onISCNCallback, false);
+  } catch (error) {
+    console.error('error occured when submitting ISCN:');
+    console.error(error);
+    lcPostInfo.arweaveAndISCNUploadStatus = 'failed';
+  }
 }
 
+async function onLikePayCallback(event) {
+  event.preventDefault();
+  if (event.origin !== 'https://like.co') { // For development, skip this line.
+    return;
+  }
+  const { action, data } = JSON.parse(event.data);
+  if (action !== 'TX_SUBMITTED') {
+    lcPostInfo.arweaveAndISCNUploadStatus = 'failed';
+    return;
+  }
+  lcPostInfo.arweaveAndISCNUploadStatus = 'uploading to Arweave...';
+  await uploadToArweave(data);
+  await onSubmitToISCN();
+}
+async function onEstimateAndUploadArweave(e) {
+  e.preventDefault();
+  lcPostInfo.mainStatusTitle = 'READY to #DePub';
```

Comment:
How can we handle i18n of this/these string?

Reply:
will pass in as lcPostInfo, hence do i18n in php as discussed